ui/store/actions.ts

```diff
@@ -976,95 +1239,98 @@ export function updateAndApproveTx(txData, dontShowLoadingIndicator) {
   };
 }
 
-export async function getTransactions(filters = {}) {
-  return await submitRequestToBackground('getTransactions', [filters]);
+export async function getTransactions(
+  filters: {
+    filterToCurrentNetwork?: boolean;
+    searchCriteria?: Partial<TransactionMeta> & Partial<TxParams>;
+  } = {},
+): Promise<TransactionMeta[]> {
+  return await submitRequestToBackground<TransactionMeta[]>('getTransactions', [
+    filters,
+  ]);
 }
 
-export function completedTx(id) {
-  return (dispatch, getState) => {
-    const state = getState();
-    const {
-      unapprovedTxs,
-      unapprovedMsgs,
-      unapprovedPersonalMsgs,
-      unapprovedTypedMessages,
-      network,
-      provider: { chainId },
-    } = state.metamask;
-    const unconfirmedActions = txHelper(
-      unapprovedTxs,
-      unapprovedMsgs,
-      unapprovedPersonalMsgs,
-      unapprovedTypedMessages,
-      network,
-      chainId,
-    );
-    const otherUnconfirmedActions = unconfirmedActions.filter(
-      (tx) => tx.id !== id,
-    );
+export function completedTx(
+  txId: number,
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return (dispatch) => {
```

Comment:
Sure! 

More context: https://github.com/MetaMask/metamask-extension/pull/17446/files/c32565e3c829c545d3a588053d809e7460a2dd6a#diff-ec46948cbcd32a43786d44f2477f99b7ebd2ab91eab6df44b10df93876e127aa

When I rebased your work on the txHelper conversion I got an error here because we were misusing it. The network and chainId should be the 7th and 8th prop but we are passing them as the 5th and 6th. So i investigated what bugs this might cause...turns out, none...because we only use the removed code to get a count of *other* unapproved transactions. in the app reducer we use this count to determine if we should set the 'shouldClosed' boolean in state. Presumably we once used this state key to decide if a popup window should close (if there are other pending messages in a popup we just move to the next if there are none the popup closes) however this logic is now all handled in the home component. shouldClose is never accessed. So any bugs we might have had are irrelevant because this is essentially a wasteful operation. Not dead cause it is still ran but its result is not used. We DO use appState.txId, so we still need this action to clear that value when the user confirms or rejects the active conf.

Reply:
thank you, super clear !